CommonTools/CandAlgos/plugins/CandPtrProjector.cc

```diff
@@ -48,6 +48,8 @@ void CandPtrProjector::produce(edm::StreamID, edm::Event& iEvent, edm::EventSetu
     auto const c = cands->ptrAt(i);
     if (vetoedPtrs.find(c) == vetoedPtrs.cend()) {
       bool addcand = true;
+      if (c->numberOfSourceCandidatePtrs() == 1)
+        addcand = (vetoedPtrs.find(c->sourceCandidatePtr(0)) == vetoedPtrs.cend());
```

Comment:
I have doubts that this check should be hardcoded.
Instead of a less debatable case of a copy, which led to this PR, some other reasons may exist for a candidate to have a source Ptr which should not be checked for a match with the veto.

smth like ```extendVetoBySingleSourcePtr``` may be a good name for this flag.
Having it would also allow for a cleaner backport to 10_6_X, if it's done.